src/hapi/pipelines/database/humanitarian_needs.py

```diff
@@ -22,147 +23,112 @@ def __init__(
-                    # "#*+idps" "#*+refugees"
-                    population_group = _get_population_group(column)
-                    # "#*+wsh" "#*+pro_gbv"
-                    sector_code = match_column(
-                        column, self.sector_pattern_to_code
-                    )
-                    if not sector_code:
-                        sector_code = "*"
-                    sector_code = sector_code.upper()
-                    gender = _get_gender(column)
-                    # "#*+age0_4" "#*+age80plus"
-                    age_range = _get_age_range(hxl_tag)
-                    min_age, max_age = _get_min_and_max_age(age_range)
-                    # "#*+disabled"
-                    disabled_marker = _get_disabled_marker(column)
-                    # TODO: Will there be columns for able bodied?
-                    for admin_code, value in values.items():
-                        try:
-                            value = int(value)
-                        except (ValueError, TypeError):
-                            continue
-                        admin2_code = admins.get_admin2_code_based_on_level(
-                            admin_code=admin_code, admin_level=admin_level
+        reader = Read.get_reader("hdx")
+        configuration = Configuration(hdx_site="stage", hdx_read_only=True)
+        configuration.setup_session_remoteckan()
+        datasets = reader.search_datasets(
+            filename="hno_dataset",
+            fq="name:hno-data-for-*",
+            configuration=configuration,
+        )
+        for dataset in datasets:
+            self._metadata.add_dataset(dataset)
+            countryiso3 = dataset.get_location_iso3s()[0]
+            time_period = dataset.get_time_period()
+            time_period_start = time_period["startdate_str"]
+            time_period_end = time_period["enddate_str"]
+            resource = dataset.get_resource()
+            resource_id = resource["id"]
+            url = resource["url"]
+            headers, rows = reader.get_tabular_rows(url, dict_form=True)
+            # Admin 1 PCode,Admin 2 PCode,Sector,Gender,Age Group,Disabled,Population Group,Population,In Need,Targeted,Affected,Reached
+            for row in rows:
+                admin2_ref = self.get_admin2_ref(countryiso3, row)
+                if not admin2_ref:
+                    continue
+                population_group = row["Population Group"]
+                if population_group == "ALL":
+                    population_group = "*"
+                sector = row["Sector"]
+                sector_code = self._sector.get_sector_code(sector)
+                if not sector_code:
+                    logger.warning(f"Sector {sector} not found!")
+                    continue
+                gender = row["Gender"]
+                if gender == "a":
+                    gender = "*"
+                age_range = row["Age Range"]
```

Comment:
I think this will need to be converted to "*"? Pending discussion [here](https://humdata.slack.com/archives/C06V0FJUZHV/p1715765530567069)

Reply:
Thanks for letting me know. Will put in a separate PR